src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -912,7 +912,9 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
         {
             path_to_size_file[index] = (char)(48 + i);
 
-            if (ReadMemoryValueFromFile(path_to_size_file, &size))
+            // Only accept reading cache sizes from size files if they are 
+            // non-bogus values i.e., non-zero and != SIZE_MAX.
+            if (ReadMemoryValueFromFile(path_to_size_file, &size) && ((size != SIZE_MAX) || (size != 0)))
```

Comment:
Do we need to keep checking for the size at all the places? Wouldn't it be sufficient to do it in the macro? It doesn't seem the `ReadMemoryValueFromFile` would ever return the `size` equal to `SIZE_MAX` and if it returned 0, it should not cause any trouble.